code-generator/src/lib.rs

```diff
@@ -9,6 +9,17 @@ use std::path::Path;
 
 pub mod catalog;
 
+const KUBERNETES_MAJOR_VERSION: usize = 33;
+
+pub fn k8s_openapi_kubernetes_version() -> String {
+    format!("v1_{}", KUBERNETES_MAJOR_VERSION)
+}
+
+pub fn kube_rs_major_version() -> String {
+    // Kubernetes was at 1.33 when kube-rs 1.0.0 was released
+    format!("{}", KUBERNETES_MAJOR_VERSION - 32)
+}
+
```

Comment:
beautiful 🤣

Reply:
oh and while this might not be the best place to discuss this, the above change also implies that the crates published from this repo will use the same major version as kube-rs. So once you release 2.x all crates here will start using 2.x as well. I hope this will make it easier for users to get the right version of the various crates they might use..

Reply:
https://github.com/metio/kube-custom-resources-rs/blob/main/code-generator/src/templates/release-crd.yml.hbs#L26 is the template for the release pipelines which now re-uses the kube-rs major version

Reply:
That seems reasonable to me.

Maybe at some point in the future we can avoid bumping majors quite so frequently (if we can somehow limit the struct crate bumping semver incompatible versions), but this is where we are at the moment.